tools/clang/lib/Sema/SemaDecl.cpp

```diff
@@ -776,11 +776,18 @@ Sema::ClassifyName(Scope *S, CXXScopeSpec &SS, IdentifierInfo *&Name,
   bool IsFilteredTemplateName = false;
   
 Corrected:
+  // Allow namespace ::dx to obtain a ULE, which will be used to construct the
+  // actual intrinsic declaration in AddOverloadedCallCandidates.
+  NamespaceDecl *nsDecl =
+      SS.isValid() ? SS.getScopeRep()->getAsNamespace() : nullptr;
+  bool isDxNamespace = nsDecl && nsDecl->getIdentifier() &&
```

Comment:
This code change is concerning. Needing custom name classification rules for the `dx` namespace seems wrong.

What declaration are you hitting that got you here?

Reply:
Removed in [2bc823b](https://github.com/microsoft/DirectXShaderCompiler/pull/7097/commits/2bc823b0df962d343ea43e8be1a6246c67474f70). 'dx' builtins now implemented the same as 'vk' ones.